src/Controllers/ShareDraftController.php

```diff
@@ -120,21 +113,32 @@ public function preview(HTTPRequest $request)
     }
 
     /**
-     * @return DBHTMLText
+     * @param string $url
+     *
+     * @return HTTPResponse
      */
-    protected function errorPage()
+    protected function getRenderedPageByURLSegment($url)
     {
-        Requirements::css('silverstripe/sharedraftcontent: client/dist/styles/bundle-frontend.css');
-        return $this->renderWith('ShareDraftContentError');
+        $pageRequest = HTTPRequestBuilder::createFromEnvironment();
+        $pageRequest->setHttpMethod('GET');
+        $pageRequest->setUrl($url);
+
+        $response = Director::singleton()->handleRequest($pageRequest);
+        if ($response->isRedirect()) {
+            // The redirect will probably be Absolute URL so just want the path
+            $newUrl = parse_url($response->getHeader('location'), PHP_URL_PATH);
```

Comment:
what happens if the redirect is to a different website?

Reply:
I'll do some tests and find out

Reply:
I've tested this locally, $newUrl will end up null so what'll happen is the share draft link will simply render the homepage.  Given redirector pages are all broken right, have it only broken for external links isn't a big deal.  Also I'm not sure what the UX should even look like since we probably wouldn't want to preview the external site.